### ISSUE TITLE
[App]: Duplicated runs with multiple sweep agents

### ISSUE THREAD

**Issue by mimxrt** [wandb/wandb#5215]
### Current Behavior

I am running 4 sweep agents in parallel and all four initial runs receive the very same parameters from the parameter list.

### Expected Behavior

Each agent should receive a different set of parameters.

### Screenshots

![grafik](https://user-images.githubusercontent.com/53339396/226968533-096d1680-e4ba-4a57-b4f0-f55f26a36e13.png)
![grafik](https://user-images.githubusercontent.com/53339396/226969227-b013e708-6a0e-433a-ac6c-2f05dd3a8c7f.png)

### Environment

OS: Server is running in Docker, OS is Ubuntu Server LTS

Browsers: Microsoft Edge

Version: 
- Server: local 0.30.0
- Client: pip 0.13.10

**Comment by mimxrt** [comment]
My guess is that using the `bayes` strategy, you are generating parameters for each run based on the results of the existing runs. Since for all four agents no runs exist at first, all four get the same set of parameters. It kind of makes sense in general but in this case it creates redundancy. If this turns out to be the case, I suggest to seed the initial parameters based on the agent index or agent identifier or something.

**Comment by MBakirWB** [comment]
Hi @mimxrt thank you for reporting, we'll investigate. Could you provide me a sample script of how you are using sweeps. Thanks

**Comment by mimxrt** [comment]
Hi @MBakirWB, thanks for your reply. I have created the following MWE for you. You can run it once to generate a sweep id and then again with the `--sweep` command line argument.

```
import sys
import time
import wandb
import wandb.sdk
import numpy as np

SWEEP_CONFIG = {
    "method": "random",
    "name": "my_config",
    "metric": {"goal": "minimize", "name": "val_root_mean_squared_error"},
    "parameters": {
        "param1": {"values": [8, 16, 32]},
        "param2": {"values": [1, 2, 4]},
    },
}


class Experiment:
    def __init__(self) -> None:
        self.x_train = np.random.random((2048, 3, 1))
        self.y_train = np.random.random((2048, 1))

    def train(self) -> None:
        run = wandb.init()
        assert run is not None
        assert type(run) is wandb.sdk.wandb_run.Run

        param1 = wandb.config.param1
        param2 = wandb.config.param2

        # sleep so that we have time to create multiple agents
        time.sleep(60)

        rmse = np.random.random(1)
        wandb.log({"val_root_mean_squared_error": rmse})
        wandb.finish()


if __name__ == "__main__":
    try:
        argi = sys.argv.index("--sweep")
        sweep_id = sys.argv[argi + 1]
        print(f"Sweep ID from cmd args: {sweep_id}")
    except Exception:
        sweep_id = wandb.sweep(sweep=SWEEP_CONFIG, project="my_proj")

    exp = Experiment()

    wandb.agent(
        sweep_id=sweep_id,
        function=exp.train,
        project="my_proj",
        count=1,
    )
```

Interestingly, this behavior is also happening for the `random` method. It seems that no matter what - the first run of each agent will always get the very same set of parameters. Here, you can see what happens when running multiple agents, each performing only one run (`count=1`):

![grafik](https://user-images.githubusercontent.com/53339396/228610507-bada15e7-110d-4a2b-90d5-82ad0147da4e.png)

**Comment by mimxrt** [comment]
This could be related to #5224. I tested with `count=50`, `method=random` and only a single agent (remove the sleep in `train()` for speed) and always the same parameters are given. 

With `method=bayes` parameters change after a few runs but the first run is always the same as described above.

**Comment by luckyhug** [comment]
In my case, I found that the sweep agents' parameters are not passing to the training script. I set my sweep epochs min:320, max:600 to avoid the default training script settings. But every time the training epochs is set to 300, which is just the same as in the default training script: 

(group.add_argument('--epochs', type=int, default=300, metavar='N',
                   help='number of epochs to train (default: 300)'))

**Comment by mimxrt** [comment]
Not sure if that has anything to do with this issue.

I just noticed that with `method=bayes` even after many runs, a lot of duplicated runs are generated. Here is a list of the most recent runs, and there are duplicates all over the place. It seems that sweeps and/or multi-agent sweeps are completely broken right now.

![grafik](https://user-images.githubusercontent.com/53339396/228844115-30eb2c42-be5a-480d-9ce5-226c93a1b815.png)

**Comment by MBakirWB** [comment]
Hi @luckyhug @mimxrt , we appreciate you reporting this and @mimxrt for providing the reproducible example. We identified this as a bug within `Server: local 0.30.0` and have a fix in place that will be released this week. In the meantime, please downgrade to version `0.29.0`.

**Comment by MBakirWB** [comment]
Hi @mimxrt, our[ latest server release](https://github.com/wandb/server/releases/tag/0.31.1) addresses this issue. Thanks again flagging it with out team. Please do reach out again anytime we could be of help.
